### PR TITLE
surf-1039-build-drag-drop-zone

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: surf-1039

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 0670748c3b74f64789bb355f99c5e2ba7ccc226a

https://app.netlify.com/sites/helix-ui/deploys/5b915907c6aed60d0fbef1b4

File: src/helix-ui/styles/elements/hx-drop-zone.less [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  &[drag="over"] {
+    &::after {
+      background-color: rgba(255, 255, 255, 0.6);
+      border-style: solid;
+    }
+  }
+  
+  &:not(.hxFence) {
+    padding: 0 !important;
```
Comment (100stacks) [inline src/helix-ui/styles/elements/hx-drop-zone.less]: move this to the top of the parent block

**Comment by araiford** [comment]
Cathy, this is really awesome!  So cool to see this interaction coming together!

A few notes below.  Please forgive, and let me know, if I'm making notes on something that was decided which I'm not aware of.

## Drop Fence

I noticed that there is no state change when you drag over the "Drop Fence" example.  I think the intent was that when the fence is the drop zone, the fence dash gets hidden and the drop-zone interaction takes it's place.  I've pasted an image from our sketch file below to illustrate.

image

## A couple notes about the drop zone and fence:

- When the fence is the immediate child of the drop zone, we should probably hide the fence on mouse drag over window, similar to the note above about the drop fence.
- It would be nice if the example for the drop zone demonstrated how the drop zone can wrap a parent container to provide a visual indicator of the larger area.  Can we add a little bit of content around the zone to demonstrate?
- I pulled the branch down and tinkered with wrapping the hx-dropzone around different elements, and I noticed that it removes the padding of the immediate child.  I did see the note about that on the page, but am not sure why we decided it has to behave that way.  Is there any way to avoid that?

**Comment by 100stacks** [comment]
Zoom LGTM

Great work @cathysiller!!! 🎉💯

**Comment by araiford** [comment]
Sorry, just now getting back to this.  I'm still seeing the Drop Fence example when I do a pull.  I think we planned to remove the Drop Fence example, since this would never be implemented without both the fence and the zone.  Is that right?

**Comment by araiford** [comment]
Thanks Cathy!  :shipit:

**Pull request merged**